plugins/module_utils/http_token.py

```diff
@@ -0,0 +1,59 @@
+from os import access, error
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+import json
+import os
+import datetime
+import requests
+import http.client
+from requests.structures import CaseInsensitiveDict
+from ansible.parsing.vault import *
+from ansible.constants import DEFAULT_VAULT_ID_MATCH
+def refresh_token(auth):
```

Comment:
I'd probably add an underscore to this and make a higher level refresh which also stores it.